dashboard/lib/sqs/queue_processor_config.rb

```diff
@@ -0,0 +1,73 @@
+require_relative '../../../lib/dynamic_config/dcdo'
+
+module SQS
+
+  # Immutable configuration for a queue processor.
+  class QueueProcessorConfig
+    attr_reader :queue_uri, :handler, :num_workers_per_processor, :num_processors
+    attr_reader :initial_max_rate, :dcdo_max_rate_key
+    attr_reader :max_wait_time, :max_batch_size
+
+    # @param [String] queue_uri URI to the SQS queue.
+    # @param [Integer] initial_max_rate Initial max rate (in messages per second) or 0 for no limit.
+    # @param [String] dcdo_max_rate_key DCDO key for the dynamic max rate.
+    # @param [Logger] logger
+    # @param [Integer] num_processors The number of processor instances.
+    # @param [Integer] num_workers_per_processor How many worker threads for each processor.
+    # @param [Integer] max_wait_time The maximum time for a long poll to wait for messages.
+    # @param [Integer] max_batch_size The maximum messages per batch. Must be <= 10.
+    def initialize(queue_uri:,
+                   handler:,
+                   initial_max_rate:,
+                   dcdo_max_rate_key:,
+                   num_processors:,
+                   num_workers_per_processor:,
+                   logger: Logger.new(STDOUT),
+                   max_wait_time: 10,
+                   max_batch_size: 10)
+
+      raise ArgumentError, 'max batch size must be <= 10' unless max_batch_size <= 10 and max_batch_size > 0
```

Comment:
Nit: I think we've been avoiding `and` because it has weird operator precedence compared to `&&`.

http://stackoverflow.com/a/2083630